app/controllers/Spaces.java

```diff
@@ -96,4 +102,114 @@ public String contentType() {
+    public static Result createSpaceFields(@ApiParam(name = "sid", value = "Space ID") Long sid) {
+        final Form<CustomFieldDefinition> newCustomFieldDefinitionForm = CUSTOM_FIELD_DEFINITION_FORM
+                .bindFromRequest();
+        User creator = User.findByAuthUserIdentity(PlayAuthenticate.getUser(session()));
+        ResourceSpace resourceSpace = ResourceSpace.read(sid);
+        if (resourceSpace == null) {
+            return notFound(Json
+                    .toJson(new TransferResponseStatus("No resource space found with id "+sid)));
+        } else {
+            if (newCustomFieldDefinitionForm.hasErrors()) {
+                TransferResponseStatus responseBody = new TransferResponseStatus();
+                responseBody.setStatusMessage(Messages.get(
+                        "The form has the following error: "+
+                                newCustomFieldDefinitionForm.errorsAsJson()));
+                return badRequest(Json.toJson(responseBody));
+            } else {
+                CustomFieldDefinition newCustomFieldDefinition = newCustomFieldDefinitionForm.get();
+                newCustomFieldDefinition.setContextUserId(creator.getUserId());
+                newCustomFieldDefinition = CustomFieldDefinition.create(newCustomFieldDefinition);
+                resourceSpace.getCustomFieldDefinitions().add(newCustomFieldDefinition);
+                resourceSpace.update();
+                return ok(Json.toJson(newCustomFieldDefinition));
+            }
+        }
+    }
+
+    @ApiOperation(httpMethod = "DELETE", response = Component.class, produces = "application/json", responseContainer = "List", value = "Delete a custom field definition from a resource space")
+    @ApiResponses(value = { @ApiResponse(code = 404, message = "No resource space found", response = TransferResponseStatus.class) })
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "SESSION_KEY", value = "User's session authentication key", dataType = "String", paramType = "header") })
+    public static Result deleteSpaceFields(@ApiParam(name = "sid", value = "Space ID") Long sid,
+                                           @ApiParam(name = "cfid", value = "Custom field definition ID") Long cfid) {
+        ResourceSpace resourceSpace = ResourceSpace.read(sid);
+        List<CustomFieldDefinition> customFieldDefinitions;
+        if (resourceSpace == null) {
+            return notFound(Json
+                    .toJson(new TransferResponseStatus("No resource space found with id "+sid)));
+        } else {
+            CustomFieldDefinition customFieldDefinition = CustomFieldDefinition.read(cfid);
+            if (customFieldDefinition == null) {
+                return notFound(Json
+                        .toJson(new TransferResponseStatus("No custom field definition found with id "+cfid)));
+            }
+            resourceSpace.getCustomFieldDefinitions().remove(customFieldDefinition);
```

Comment:
For the delete, let's simply make a "soft" delete by updating the field "removed" to "TRUE" in the custom field entity